Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -52,11 +63,25 @@ public class ApolloCodegen {
       compilationResult: compilationResult
     )
 
+    var existingGeneratedFilePaths = configuration.options.pruneGeneratedFiles ?
+    try findExistingGeneratedFilePaths(
+      config: configContext,
+      fileManager: fileManager
+    ) : []
+
     try generateFiles(
       compilationResult: compilationResult,
       ir: ir,
-      config: configContext
+      config: configContext,
+      fileManager: fileManager
     )
+
+    if configuration.options.pruneGeneratedFiles {
+      try deleteExtraneousGeneratedFiles(
+        from: &existingGeneratedFilePaths,
+        afterCodeGenerationUsing: fileManager
```

Comment:
It was just to make it clear that this is very much dependent on you doing this AFTER generating code and with the same file manager you used for generation.

If you moved this function up above `generateFiles`, it would run just fine but wouldn't do what we want it to exactly. (You'd actually end up with the same outcome I think, just all files would get deleted then re-created.)

I'm not super hung up on the name, but I think it's more descriptive of a piece of implicit knowledge and implementation details than `fileManager`.

Reply:
I get why you named it that, but disagree on the value of it. I'm not concerned enough about it to request a change, this is fine.